edit.js

```diff
@@ -1,44 +1,125 @@
+let company = document.getElementById("company");
+let startDate = document.getElementById("start-date");
+let endDate = document.getElementById("end-date");
+let description = document.getElementById("description");
+
+let isEdit = false;
+
 function showExperienceForm() {
   document.getElementById("experience-form").style.display = "block";
 }
 document.getElementById("experience-form").addEventListener("submit", function (event) {
+  if (!company.value || !startDate.value || !endDate.value || !description.value) {
     event.preventDefault();
+    return;
+  }
+
+  event.preventDefault();
+  
+    let experiences = JSON.parse(localStorage.getItem("experience")) || [];
+
+    let id = Date.now().toString(36) + Math.random().toString(36).substr(2, 9);
+
+    if (isEdit) {
+      let newExperience = {
+        id: isEdit,
+        company: company.value,
+        startDate: startDate.value,
+        endDate: endDate.value,
+        description: description.value,
+      };
 
-    let company = document.getElementById("company");
-    let startDate = document.getElementById("start-date");
-    let endDate = document.getElementById("end-date");
-    let description = document.getElementById("description");
-    
-    if (
-      company.value &&startDate.value &&endDate.value &&description.value ) {
-      let experiences = JSON.parse(localStorage.getItem("experience")) || [];
+      let newExperiences = experiences.map((exp) => {
+        if (exp.id == isEdit) {
+          return newExperience;
+        } else {
+          return exp;
+        }
+      });
+
+      localStorage.setItem("experience", JSON.stringify(newExperiences));
+      isEdit = false;
+    } else if (company.value && startDate.value && endDate.value && description.value) {
       experiences.push({
+        id: id,
         company: company.value,
         startDate: startDate.value,
         endDate: endDate.value,
         description: description.value,
       });
-      document.getElementById("experience-form").reset();
+
       localStorage.setItem("experience", JSON.stringify(experiences));
-      document.getElementById("experience-form").style.display = "none";
-      loadExperiences();
     }
+
+    document.getElementById("experience-form").reset();
+    document.getElementById("experience-form").style.display = "none";
```

Comment:
use the same approach as you have used to show like showExperienceForm.

Reply:
Done